nixos/modules/virtualisation/qemu-guest-agent.nix

```diff
@@ -0,0 +1,38 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.qemu-guest-agent;
+in {
+
+  ###### interface
+
+  options.services.qemu-guest-agent = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to enable the dspam spam filter.";
+      };
+  };
+
+  config = mkIf cfg.enable (mkMerge [
+    {
+
+      services.udev.extraRules = ''
+        SUBSYSTEM=="virtio-ports", ATTR{name}=="org.qemu.guest_agent.0", TAG+="systemd" ENV{SYSTEMD_WANTS}="qemu-guest-agent.service"
+      '';
+
+      environment.systemPackages = [ pkgs.qemu ];
```

Comment:
This is problematic: 

1. It makes the guest depend on full `qemu` although it just needs the guest-agent.
2. It's a hard reference to standard `qemu`. We have several variants like `qemu_test` and `qemu_kvm`, and it is difficult to tell from the guest config which one the host will use to run the vm. So this may build/download an additional `qemu` version ... just for the guest-agent.

I think the cleanest solution is to create a separate small `qemu-guest-agent` package (as many other distros do), no variants needed for this.

Reply:
I agree, in fact I had started writing the package qemu-agent before realizing it was already in the package. I am not sure how easy it is to generate just the Qemu agent, I wouldn't mind someone else doing it (or I can try later but I am short on time this month).

Reply:
thanks for the agent
Regarding your comment
> It's a hard reference to standard qemu. We have several variants like qemu_test and qemu_kvm, and it is difficult to tell from the guest config which one the host will use to run the vm. So this may build/download an additional qemu version ... just for the guest-agent.

How do you propose to solve it ?
Should I use another option qemuPackage ?

Reply:
> Should I use another option qemuPackage ?

You could do that but not sure I'm not sure that's necessary. If you use `qemu.ga` in the guest config, it should only install the small `ga` output.